### DISCUSSION TITLE
Argo Rollouts UI Service and Authentication

### DISCUSSION THREAD

**Discussion by zezaeoh** [argoproj/argo-rollouts#1204] [Q&A] [answered]
Hi, there!

I wonder if there is any plan to export argo rollout ui to service and add authentication feature in it.

We are testing argo rollout's action ui with argo cd.
but, argo cd is not enough to understand which step user paused and how many traffic goes in

**Comment by alwaysastudent** [comment]
Does argo rollout provide a cluster deployable web UI?

**Comment by jessesuen** [comment, marked as answer]
I agree the project should improve to provide a set of kubernetes manifests for the rollouts dashboard so that it can be exposed with a service. 

We are actively thinking about how to integrate rollout dashboard into argo-cd in a loosely coupled way:
* argoproj/argo-cd#6240
* argoproj/argo-cd#6232

**Reply by nicolasfagotti** [reply]
Hi @zezaeoh and @jessesuen! Is this being followed up anywhere else? I see the code for the extension was not merged, so I assume the decision was to follow a different direction than the proposed but I couldn't find anything yet

**Reply by deadlytea** [reply]
Just in case someone like myself is looking where this has been implemented, it's [here](https://github.com/argoproj/argo-cd/issues/5432) and available as of ArgoCD v2.1.5. [Link](https://github.com/argoproj-labs/rollout-extension) to rollouts extension.

**Reply by Hronom** [reply]
@deadlytea it seems like extension not well supported, it will be good to have it as part of official rollouts of Argo Rollouts project. So this extension will be more up-to date and have a proper support.

Maybe integrate directly in Argo CD and make a possibility to enable it by some property.
